### PR TITLE
BE Setup for Users list

### PR DESCRIPTION
**Pull request by iraboldyrieva**
## Describe your changes

- Filtering by roles.
- Sorting by multiple fields.
- Pagination for user navigation.
- User list retrieval endpoint integrated with company-level restrictions.

## Issue ticket code (and/or) and link

- [SCRUM-28](https://codecraftersintership.atlassian.net/browse/SCRUM-28)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

-[x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [ ] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

### CODE REVIEW

File: src/common/guards/userCompany.guard.ts [hunk 1]
```diff
@@ -7,37 +7,54 @@ import { Request } from 'express';
...
+
+    const authorizationHeader = request.headers.authorization;
+    if (!authorizationHeader) {
+      throw new UnauthorizedException('Authorization token is missing');
+    }
+
+    const decodedToken = jwt.decode(authorizationHeader);
+    if (typeof decodedToken !== 'object' || decodedToken === null) {
+      throw new UnauthorizedException('Invalid token');
+    }
```
Comment (lyudad) [inline src/common/guards/userCompany.guard.ts]: откуда эта идея что то доставать с токена, токен нужен беку. фронт данные берет с редакса или с запроса

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -1,9 +1,8 @@
-import { Controller, Get, Query, SetMetadata, UseGuards, Req, BadRequestException } from '@nestjs/common';
+import { BadRequestException, Controller, Delete, Get, Param, Query, Req, SetMetadata, UseGuards } from '@nestjs/common';
 import { ApiBearerAuth, ApiOperation, ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { Roles } from 'common/enums/enums';
-import { AuthGuard } from 'common/guards/auth.guard';
 import { RolesGuard } from 'common/guards/roles.guard';
-import { UserCompanyGuard, AccessTokenInterface } from 'common/guards/userCompany.guard';
+import { AccessTokenInterface, UserCompanyGuard } from 'common/guards/userCompany.guard';
 import { Request } from 'express';
```
Comment (Parle648) [inline src/modules/user/user.controller.ts:6]: 1.1.14

**Review by Parle648 — COMMENTED** [review]
left comments

**Review by Parle648 — APPROVED** [review]

File: src/modules/user/driver/driver.service.ts [hunk 3]
```diff
@@ -80,6 +80,8 @@ export class DriverService {
     try {
       const dispatcher = await this.userRepo.findOneByOrFail({ id });
 
+      console.log(dispatcher);
```
Comment (oyakovytskyi) [inline src/modules/user/driver/driver.service.ts]: 1.1.8

**Review by oyakovytskyi — APPROVED** [review]

File: src/modules/user/driver/driver.service.ts (hunk 3 above)
Comment (Parle648) [reply src/modules/user/driver/driver.service.ts]: done

**Review by AlexeyBezrebriy — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**